packages/app-builder/src/components/Decisions/RulesDetail.tsx

```diff
@@ -75,35 +78,23 @@ export function RulesDetail({
                 </Accordion.Title>
                 <Accordion.Content className="bg-purple-02 border-grey-10 border-t">
                   <div className="flex flex-col gap-4 p-4">
-                    {rule.description ? (
-                      <Callout variant="outlined">{rule.description}</Callout>
-                    ) : null}
-                    <div className="bg-purple-10 inline-flex h-8 w-fit items-center justify-center whitespace-pre rounded px-2 font-normal text-purple-100">
-                      <Trans
-                        t={t}
-                        i18nKey="scenarios:rules.consequence.score_modifier"
-                        components={{
-                          Score: <span className="font-semibold" />,
-                        }}
-                        values={{
-                          score: formatNumber(rule.ruleDetail.scoreModifier, {
-                            language,
-                            signDisplay: 'always',
-                          }),
-                        }}
-                      />
-                    </div>
-                    {rule.ruleDetail.formula ? (
-                      <RuleFormula
-                        formula={rule.ruleDetail.formula}
-                        databaseAccessors={databaseAccessors}
-                        payloadAccessors={payloadAccessors}
-                        astOperators={astOperators}
-                        dataModel={dataModel}
-                        customLists={customLists}
-                        triggerObjectType={triggerObjectType}
-                      />
+                    {ruleExecution.description ? (
+                      <Callout variant="outlined">
+                        {ruleExecution.description}
+                      </Callout>
                     ) : null}
+
+                    <Suspense fallback={t('common:loading')}>
```

Comment:
just for my understanding: why does this need to be done async ? There's no network call in the component, right ?

Reply:
If you look at the component props declaration, we pass `Promise<....>` for astRuleData (this is due to the `defer` usage).
`Suspense` allow to handle asynchronous variables, displaying the `fallback` until promises resolve

Reply:
**Exception :**
This do not protect against exception like `try ... catch` block. 
If an exception is raised, this trigger the classic behaviour in this case : display the exported (if any) `ErrorBoundary` component from the page (or bubble up if none)

**Await :**
This king of "replace" the `await` statement.
When a navigation/transition occure in Remix, it `await` for all required loaders to resolve before doing anuthing (we already have some latency due to this in some part of the app in staging/prod due to network + real DB). 
Using a `defer` return statement allow to mix data we require to display the page and data we "defer" :
- required data = data needed to display something valuable at first (preventing to have loaders everywhere)
- deferred data = data not needed at first paint or data that take a huge amount of time to fetch

Here we have data that take some time and that is only used in accordion contents (hidden at first) so it makes sense to `defer` it.

ATM, if the promises reject, the page will transition from "decision content" to "error page". If it occures a lot, we can try/catch the error and return a possibly undefined value (= display an error message only where the data is used).
An alternative would be to move the rule detail inside a dedicated `Outlet` and handle errors the classic way with Remix (so the error component will be only displayed in place of the `Outlet` in the final layout)